### PR TITLE
fix(netlify): avoid shipping two binaries

### PR DESCRIPTION
**Pull request by millsp**
possible fix for #6503

### CODE REVIEW

**Review by janpio — CHANGES_REQUESTED** [review]
Includes unrelated code changes.

**Comment by Jolg42** [comment]
@janpio merged master, unrelated changed is gone

**Review by janpio — CHANGES_REQUESTED** [review]
Can we do this as an integration branch that gets a release please, so we can test his in our e2e tests? 
I want to confirm this working end to end first.

**Comment by Jolg42** [comment]
I see that it was published with version `2.21.0-integration-netlify-binary.1`

**Comment by Jolg42** [comment]
And e2e are green for this version https://github.com/prisma/e2e-tests/actions/runs/741302308

**Comment by janpio** [comment]
The current e2e tests do not fail based on this. I need to do manual stuff.

**Review by janpio — CHANGES_REQUESTED** [review]
Unfortunately I can not confirm this working yet:

#### Old version `2.21.0-dev.21`:
https://github.com/prisma/e2e-tests/pull/1613
https://github.com/prisma/e2e-tests/pull/1613/checks?check_run_id=2315077041
```
"files":["index-browser.js","index.d.ts","index.js","package.json","query-engine-debian-openssl-1.0.x","query-engine-rhel-openssl-1.0.x","schema.prisma"]
```

#### New version `2.21.0-integration-netlify-binary.1`:
https://github.com/prisma/e2e-tests/pull/1618
https://github.com/prisma/e2e-tests/pull/1618/checks?check_run_id=2324984121
```
"files":["index-browser.js","index.d.ts","index.js","package.json","query-engine-debian-openssl-1.0.x","query-engine-rhel-openssl-1.0.x","schema.prisma"]
```

So right now it seems that the `query-engine-debian-openssl-1.0.x` is still uploaded.

Reproduction with https://github.com/janpio/prisma-binary-repro might be easier than using e2e test for debugging though.

**Comment by janpio** [comment]
Also created a PR for my `prisma-binary-repro`:

- `2.21.0-dev.21`: https://prisma-binary-repro.netlify.app/.netlify/functions/index
- `2.21.0-integration-netlify-binary.1`: https://deploy-preview-2--prisma-binary-repro.netlify.app/.netlify/functions/index

Both show the same binaries uploaded to AWS Lambda :(

File: src/packages/client/src/generation/generateClient.ts [hunk 1]
```diff
@@ -246,9 +246,14 @@ export async function generateClient({
     )
   }
   if (transpile) {
-    for (const filePath of Object.values(enginePath)) {
+    if (process.env.NETLIFY) await makeDir('/tmp/prisma-engines')
```
Comment (Jolg42) [inline src/packages/client/src/generation/generateClient.ts:249]: I'm surprised prettier/linting didn't do anything, we usually prefer 
```suggestion
    if (process.env.NETLIFY) {
    	await makeDir('/tmp/prisma-engines')
    }
```

File: src/packages/client/src/generation/generateClient.ts [hunk 2]
```diff
@@ -246,9 +246,22 @@ export async function generateClient({
...
+    }
+
+    for (const [binaryTarget, filePath] of Object.entries(enginePath)) {
+      console.warn(
+        `.env: ${process.env.NETLIFY}`,
+        `binary: ${binaryTarget}`,
+        `path: ${filePath}`,
+      )
+
       const fileName = path.basename(filePath)
```
Comment (Jolg42) [inline src/packages/client/src/generation/generateClient.ts:260]:
```suggestion
      const fileName = path.basename(filePath)
```

I think we can now remove this @millsp ?

Comment (millsp) [reply src/packages/client/src/generation/generateClient.ts:260]: Yep, reverse my last 3 commits

File: src/packages/client/src/generation/generateClient.ts [hunk 3]
```diff
@@ -246,9 +246,22 @@ export async function generateClient({
     )
   }
   if (transpile) {
-    for (const filePath of Object.values(enginePath)) {
+    if (process.env.NETLIFY) {
+    	await makeDir('/tmp/prisma-engines')
```
Comment (Jolg42) [inline src/packages/client/src/generation/generateClient.ts:250]:
```suggestion
      await makeDir('/tmp/prisma-engines')
```

**Comment by Jolg42** [comment]
We confirmed it's fixed and we are going to merge and release a patch to include it in the current release.

**Review by Jolg42 — APPROVED** [review]

**Comment by Jolg42** [comment]
Included in `2.21.2`

**Pull request merged**